plugins/inputs/tacacs/tacacs_test.go

```diff
@@ -156,54 +150,55 @@ func TestTacacsLocal(t *testing.T) {
+			testingTimeout: config.Duration(0),
+			serverToTest:   []string{srvLocal},
+			usedUsername:   config.NewSecret([]byte(`testusername`)),
+			usedPassword:   config.NewSecret([]byte(`testpassword`)),
+			usedSecret:     config.NewSecret([]byte(`testsecret`)),
+			requestAddr:    "127.0.0.1",
+			reqRespStatus:  "AuthenStatusPass",
 		},
 		{
-			name:                 "wrongpw",
-			testingTimeout:       config.Duration(time.Second * 5),
-			serverToTest:         []string{srvLocal},
-			usedUsername:         config.NewSecret([]byte(`testusername`)),
-			usedPassword:         config.NewSecret([]byte(`WRONGPASSWORD`)),
-			usedSecret:           config.NewSecret([]byte(`testsecret`)),
-			requestAddr:          "127.0.0.1",
-			expectNoGatherErrors: true,
+			name:           "wrongpw",
+			testingTimeout: config.Duration(time.Second * 5),
+			serverToTest:   []string{srvLocal},
+			usedUsername:   config.NewSecret([]byte(`testusername`)),
+			usedPassword:   config.NewSecret([]byte(`WRONGPASSWORD`)),
+			usedSecret:     config.NewSecret([]byte(`testsecret`)),
+			requestAddr:    "127.0.0.1",
+			reqRespStatus:  "AuthenStatusFail",
 		},
 		{
-			name:                 "wrongsecret",
-			testingTimeout:       config.Duration(time.Second * 5),
-			serverToTest:         []string{srvLocal},
-			usedUsername:         config.NewSecret([]byte(`testusername`)),
-			usedPassword:         config.NewSecret([]byte(`testpassword`)),
-			usedSecret:           config.NewSecret([]byte(`WRONGSECRET`)),
-			requestAddr:          "127.0.0.1",
-			expectNoGatherErrors: false,
+			name:           "wrongsecret",
+			testingTimeout: config.Duration(time.Second * 5),
+			serverToTest:   []string{srvLocal},
+			usedUsername:   config.NewSecret([]byte(`testusername`)),
+			usedPassword:   config.NewSecret([]byte(`testpassword`)),
+			usedSecret:     config.NewSecret([]byte(`WRONGSECRET`)),
+			requestAddr:    "127.0.0.1",
+			errContains:    "error on new tacacs authentication start request to " + srvLocal + " : bad secret or packet",
 		},
 		{
-			name:                 "unreachable",
-			testingTimeout:       config.Duration(time.Nanosecond * 1000),
-			serverToTest:         []string{"unreachable.hostname.com:404"},
-			usedUsername:         config.NewSecret([]byte(`testusername`)),
-			usedPassword:         config.NewSecret([]byte(`testpassword`)),
-			usedSecret:           config.NewSecret([]byte(`testsecret`)),
-			requestAddr:          "127.0.0.1",
-			expectNoGatherErrors: false,
+			name:           "unreachable",
+			testingTimeout: config.Duration(time.Nanosecond * 1000),
+			serverToTest:   []string{"unreachable.hostname.com:404"},
+			usedUsername:   config.NewSecret([]byte(`testusername`)),
+			usedPassword:   config.NewSecret([]byte(`testpassword`)),
+			usedSecret:     config.NewSecret([]byte(`testsecret`)),
+			requestAddr:    "127.0.0.1",
+			errContains:    "error on new tacacs authentication start request to unreachable.hostname.com:404 : dial tcp",
```

Comment:
hostname.com is actually in use, so use one of the reserved names. Example:
```suggestion
			serverToTest:   []string{"unreachable.test:49"},
			usedUsername:   config.NewSecret([]byte(`testusername`)),
			usedPassword:   config.NewSecret([]byte(`testpassword`)),
			usedSecret:     config.NewSecret([]byte(`testsecret`)),
			requestAddr:    "127.0.0.1",
			errContains:    "error on new tacacs authentication start request to unreachable.test:49 : dial tcp",
```